### PR TITLE
Fix disappearing MasterDetail menu on UWP

### LINKED ISSUE #5412: Xamarin.Forms UWP Navigation bar Disappear while press back button to MasterDetailPage
MasterDetailPage is the root page.
Press button from master page's button ,Navigate to Setting Page is done.
Press the back button on the top of the window , Navigate to Mainpage(MasterDetailPage).
Navigation bar Disappear
(I had been set `<NavigationPage.HasNavigationBar>true</NavigationPage.HasNavigationBar>` in detail page)

Sample code here:
[TopBarSample.zip](https://github.com/xamarin/Xamarin.Forms/files/2914278/TopBarSample.zip)

### LINKED ISSUE #5637: [Feedback] Xamarin Forms Shared project UWP menu disappears with no way to go back or show the menu again.
### Description

<p><p>I implemented MasterDetailPage into my app, but on Windows 10 the behavior seems a bit weird. When the app starts, the menu is showing as expected. But when I click an item, it disappears with no way to go back or show the menu again.</p>
<p>This has been reported before some 19months ago and isn’t something new.</p>
<p><a target=_blank href="https://stackoverflow.com/questions/45640159/xamarin-masterdetailpage-menu-on-uwp">https://stackoverflow.com/questions/45640159/xamarin-masterdetailpage-menu-on-uwp</a></p>
<p>Easy to reproduce  with a MasterDetailPage set MasterBehavior = MasterBehavior.Popover</p>
<ol>
<li>
<p>Detail = …some new page;<br>
IsPresented = false;</p>
</li>
<li>
<p>. …menu is now gone forever in UWP?</p>
</li>
</ol>
</p>
<a href="https://developercommunity.visualstudio.com/content/problem/489958/xamarin-forms-shared-project-uwp-menu-disappears-w.html">View on Developer Community</a>
<br>To find diagnostic information, see the <a href="https://devdiv.visualstudio.com/DevDiv/_workitems/edit/820675">original linked feedback ticket</a>.

### Source

[823510](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/823510)

### PR DESCRIPTION
**Pull request by mrlacey**
### Description of Change ###

Fix issue where unable to get to the Popover MasterDetail menu so unable to navigate through the app.

### Issues Resolved ### 

- fixes #5412
- fixes #5637 (although not 100% certain as it doesn't have a good repro)

### API Changes ###
 
 None

### Platforms Affected ### 

- UWP

### Behavioral/Visual Changes ###

Before: gets into a state where can't navigate.
Now: doesn't get in that state.

### Before/After Screenshots ### 

Added tests which include manual instructions on how to repro the issue/verify the fix.

Before: (Navigating through the app could lead to hamburger menu not being displayed)
image

After: (Even after navigating between pages, the hamburger menu is always displayed)
image

### Testing Procedure ###

**Only tested on UWP** as the reported issue only happens there.
Recording of automated test run (launched from within VS)
Issue5412-test-run-recording

### PR Checklist ###

- [x] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

**Comment by rmarinho** [comment]
Can you run UITest on this @PureWeen ? thanks

**Comment by rmarinho** [comment]
@mrlacey i think we should make this UITests just for Windows ?

**Comment by mrlacey** [comment]
> @mrlacey i think we should make this UITests just for Windows ?

Fine by me

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue5412.cs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 5412, "5412 - (NavigationBar disappears on MasterDetailPage)", PlatformAffected.UWP)]
+	public class Issue5412 : TestContentPage
+	{
+		protected override async void Init()
+		{
+			await Navigation.PushModalAsync(new Issue5412MainPage());
+		}
+
+#if UITEST
```
Comment (samhouts) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue5412.cs]:
```suggestion
#if UITEST && __WINDOWS__
```

**Comment by mrlacey** [comment]
Are those real failing build checks? Without access to the logs, they look like temporary issues on the build server.

**Comment by mrlacey** [comment]
How can I see more information about these failing checks?

**Pull request merged**